src/extension.ts

```diff
@@ -41,6 +42,7 @@ export function activate(context: vs.ExtensionContext) {
 		return; // Don't set anything else up; we can't work like this!
 	}
 	context.globalState.update(stateLastKnownSdkPathName, dartSdkRoot);
+	debug.writeSdkPath(dartSdkRoot); // Write the SDK path for the debugger to find.
```

Comment:
We do this on startup, but we should also write the value when the preference changes.

We'll run into some subtle issues if the user has multiple copies of vscode open, with different values for the dart sdk path in each. `run` will pull the sdk value from the various copies of vscode non-deterministically. This may not be a common issue; perhaps future changes to the debugging api would let us address this.